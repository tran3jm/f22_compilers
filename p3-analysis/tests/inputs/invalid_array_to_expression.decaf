bool a[1];
def int main()
{
    a = !"test" * 42 && "ok" / 43;
    return 5;
}