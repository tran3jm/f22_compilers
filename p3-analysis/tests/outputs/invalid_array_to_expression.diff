--- outputs/invalid_array_to_expression.txt	2022-10-26 14:33:46.126565707 -0400
+++ expected/invalid_array_to_expression.txt	2022-10-26 14:32:08.485540213 -0400
@@ -1,5 +1,6 @@
 Array 'a' accessed without index on line 4
-Type mismatch: int is incompatible with str on line 4
-Type mismatch: bool is incompatible with int on line 4
-Type mismatch: str is incompatible with int on line 4
-Type mismatch: int is incompatible with int on line 4
+Type mismatch: bool expected but str found on line 4
+Type mismatch: int expected but bool found on line 4
+Type mismatch: int expected but str found on line 4
+Type mismatch: bool expected but int found on line 4
+Type mismatch: bool expected but int found on line 4
