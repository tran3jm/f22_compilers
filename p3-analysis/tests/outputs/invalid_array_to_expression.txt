Array 'a' accessed without index on line 4
Type mismatch: int is incompatible with str on line 4
Type mismatch: bool is incompatible with int on line 4
Type mismatch: str is incompatible with int on line 4
Type mismatch: int is incompatible with int on line 4
