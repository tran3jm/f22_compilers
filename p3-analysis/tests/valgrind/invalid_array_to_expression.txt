==2844274== Memcheck, a memory error detector
==2844274== Copyright (C) 2002-2017, and GNU GPL'd, by Julian Seward et al.
==2844274== Using Valgrind-3.15.0 and LibVEX; rerun with -h for copyright info
==2844274== Command: ../decaf inputs/invalid_array_to_expression.decaf
==2844274== 
Array 'a' accessed without index on line 4
Type mismatch: int is incompatible with str on line 4
Type mismatch: bool is incompatible with int on line 4
Type mismatch: str is incompatible with int on line 4
Type mismatch: int is incompatible with int on line 4
==2844274== 
==2844274== HEAP SUMMARY:
==2844274==     in use at exit: 0 bytes in 0 blocks
==2844274==   total heap usage: 1,905 allocs, 1,905 frees, 241,128 bytes allocated
==2844274== 
==2844274== All heap blocks were freed -- no leaks are possible
==2844274== 
==2844274== For lists of detected and suppressed errors, rerun with: -s
==2844274== ERROR SUMMARY: 0 errors from 0 contexts (suppressed: 0 from 0)
