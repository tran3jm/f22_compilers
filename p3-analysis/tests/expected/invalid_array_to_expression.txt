Array 'a' accessed without index on line 4
Type mismatch: bool expected but str found on line 4
Type mismatch: int expected but bool found on line 4
Type mismatch: int expected but str found on line 4
Type mismatch: bool expected but int found on line 4
Type mismatch: bool expected but int found on line 4
